### PR TITLE
Ovrd Scholarship Modal

### PR DESCRIPTION
**Pull request by lindsayrmaher**
### What's this PR do?
paired with @Dinnall !

This pull request adds the bare bones functionality for the 'view scholarship details' link to successfully open the scholarship modal with scholarship information!

### How should this be reviewed?

Should we be taking into account the possibility that this will be posted on an affiliate website site as well? Does it make sense to hard code the text and scholarship details here and pull the action/time from the campaign? I can't think of a cleaner way to pull the scholarship info since we aren't pulling scholarship info on the graphql queries right now.  

### Any background context you want to provide?

Continuing work to make the beta page a reasonable replica of a general campaign page without duplicating unnecessary code/work.

### Relevant tickets

References [Pivotal # 172439286](https://www.pivotaltracker.com/story/show/172439286).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

**Comment by lindsayrmaher** [comment]
This has a confusing amount of code changes in it since we haven't merged in the first OVRD ticket yet...

**Comment by aaronschachter** [comment]
> This has a confusing amount of code changes in it since we haven't merged in the first OVRD ticket yet...

When you have a chance, let's rebase this branch with master so it's easier to see the changes.

File: resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js [hunk 1]
```diff
@@ -69,7 +76,14 @@ const BetaVoterRegistrationDrivePage = () => {
...
   const { firstName } = data.user;
-  const { coverImage, title } = data.campaignWebsite;
+  const {
+    campaignId,
+    coverImage,
+    title,
+    scholarshipAmount,
+    scholarshipDeadline,
+    additionalContent,
+  } = data.campaignWebsite;
```
Comment (aaronschachter) [inline resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js:86]: Let's list these in alphabetical order to help readability. Note to self -- totes need to add this unspoken preference into a spoken one in our new [Contributing doc](https://github.com/DoSomething/communal-docs/tree/master/Contributing).

File: resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js [hunk 2]
```diff
@@ -79,10 +93,10 @@ const BetaVoterRegistrationDrivePage = () => {
...
   const campaignInfoBlock = (
     <CampaignInfoBlock
-      campaignId={isDevEnvironment ? 9035 : 9054}
-      hideScholarshipDetails
-      scholarshipAmount={1500}
-      scholarshipDeadline="2020-04-30"
+      scholarshipAmount={scholarshipAmount}
+      scholarshipDeadline={scholarshipDeadline}
+      showModal={() => setShowScholarshipModal(true)}
+      campaignId={campaignId}
```
Comment (aaronschachter) [inline resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js:99]: Same note here, let's keep this alphabetically listed.

File: resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js [hunk 3]
```diff
@@ -149,6 +163,22 @@ const BetaVoterRegistrationDrivePage = () => {
...
+            onClose={() => {
+              setShowScholarshipModal(false);
+            }}
+            trackingId="SCHOLARSHIP_MODAL_BETA_VOTER_REGISTRATION"
+          >
+            <ScholarshipInfoBlock
+              campaignId={campaignId}
+              scholarshipAmount={scholarshipAmount}
+              scholarshipDeadline={scholarshipDeadline}
+              numberOfScholarships={additionalContent.numberOfScholarships}
```
Comment (aaronschachter) [inline resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js:178]: Does this break if `additionalContent` is null? This is a good example of where it'd be nice to start adding Cypress tests for each of these gradual changes we're making.

File: resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js [hunk 4]
```diff
@@ -41,6 +47,7 @@ const BetaVoterRegistrationDrivePage = () => {
    * @see /docs/development/features/voter-registration
    */
   const voterRegistrationDriveCampaignWebsiteId = '3pwxnRZxociqMaQCMcGOyc';
+  const [showScholarshipModal, setShowScholarshipModal] = useState(false);
```
Comment (aaronschachter) [inline resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js:50]: 💯

File: resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js (hunk 1 above)
Comment (lindsayrmaher) [reply resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaPage.js:84]: 💯 yeeeeees!

File: resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaHeroSection.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import CoverImage from '../../../utilities/CoverImage/CoverImage';
+import CampaignInfoBlock from '../../../blocks/CampaignInfoBlock/CampaignInfoBlock';
+
+const BetaHeroSection = ({ user, campaignInfo, modalToggle }) => {
```
Comment (aaronschachter) [inline resources/assets/components/pages/VoterRegistrationDrivePage/Beta/BetaHeroSection.js:7]: Nit - can we just call this `HeroSection`? The other subcomponents besides the `AlphaPage` and `BetaPage` within the `pages/VoterRegistrationDrivePage` don't have similar prefixes.

**Pull request merged**